File: JniPMML-Java/src/main/java/com/WDataSci/JniPMML/RecordSetMDEnums.java
package com.WDataSci.JniPMML;

public class RecordSetMDEnums
{
    public enum eType
    {
        Unknown(0), Dlm(1), CSV(2), TXT(3), HDF5(4), DBB(5), XML(6), JSON(7), SQL(8);

        private final int val;

        eType(int arg)
        {
            this.val = arg;
        }

        public boolean isFlatFile()
        {
            return this.bIn(Dlm, CSV, TXT, XML, JSON);
        }

        public boolean isFile()
        {
            return this.bIn(Dlm, CSV, TXT, XML, JSON, HDF5);
        }

        public static eType FromInt(int arg)
        {
            switch (arg) {
                case 1:
                    return Dlm;
                case 2:
                    return CSV;
                case 3:
                    return TXT;
                case 4:
                    return HDF5;
                case 5:
                    return DBB;
                case 6:
                    return XML;
                case 7:
                    return JSON;
                case 8:
                    return SQL;
                default:
                    return Unknown;
            }
        }


        public static eType FromAlias(String arg)
        {
            switch (arg.toLowerCase()) {
                case "csv":
                case "comma-delimited":
                case "comma-separated":
                    return eType.CSV;
                case "txt":
                case "tab-delimited":
                case "tab-separated":
                    return eType.TXT;
                case "dlm":
                case "pipe":
                case "pipe-delimited":
                case "pipe-separated":
                case "delimited":
                    return eType.Dlm;
                case "hdf5":
                case "h5":
                    return eType.HDF5;
                case "xdatadbb":
                case "bytebuffer":
                case "bb":
                case "byte":
                case "bytes":
                    return eType.DBB;
                case "sql":
                    return eType.SQL;
                case "json":
                    return eType.JSON;
                case "xml":
                    return eType.XML;
                default:
                    return eType.Unknown;
            }
        }

        @Override
        public String toString()
        {
            switch (this) {
                case Dlm:
                    return "Dlm";
                case CSV:
                    return "CSV";
                case TXT:
                    return "TXT";
                case HDF5:
                    return "HDF5";
                case DBB:
                    return "DBB";
                case XML:
                    return "XML";
                case JSON:
                    return "JSON";
                case SQL:
                    return "SQL";
                default:
                    return "Unknown";
            }
        }

        public boolean bIn(eType... args)
        {
            for (eType arg : args) if ( this.equals(arg) ) return true;
            return false;
        }

        public boolean bIn(String... arg)
        {
            boolean found = false;
            for (int i = 0; !found && i < arg.length; i++) {
                switch (arg[i].toLowerCase()) {
                    case "dlm":
                    case "delimited":
                        if ( this.equals(eType.Dlm) ) found = true;
                        break;
                    case "csv":
                        if ( this.equals(eType.CSV) ) found = true;
                        break;
                    case "txt":
                        if ( this.equals(eType.TXT) ) found = true;
                        break;
                    case "hdf5":
                    case "h5":
                        if ( this.equals(eType.HDF5) ) found = true;
                        break;
                    case "xdatadbb":
                    case "bytebuffer":
                    case "bb":
                        if ( this.equals(eType.DBB) ) found = true;
                        break;
                    default:
                }
            }
            return false;
        }
    }

    public enum eMode
    {
        Unknown(0), Input(1), Output(2), Internal(3);
        private final int val;

        eMode(int arg)
        {
            this.val = arg;
        }

        public boolean Equals(eMode arg) {
            return this.equals(arg);
        }

        public boolean bIn(eMode... args)
        {
            for (eMode arg : args) if ( this.equals(arg) ) return true;
            return false;
        }

        @Override
        public String toString()
        {
            switch (this) {
                case Input:
                    return "InputMap";
                case Output:
                    return "OutputMap";
                case Internal:
                    return "InternalMap";
                default:
                    return "Unknown";
            }
        }
    }

    public enum eSchemaType
    {
        Unknown(0), XSD(1), XML(2), JSON(3), HDF5(4), DBB(5), RecordSetMD(6), SQL(7), NamingConvention(8);
        private final int val;

        eSchemaType(int arg)
        {
            this.val = arg;
        }

        public boolean bIn(eSchemaType... args)
        {
            for (eSchemaType arg : args) if ( this.equals(arg) ) return true;
            return false;
        }

        public static eSchemaType FromInt(int arg)
        {
            switch (arg) {
                case 1:
                    return XSD;
                case 2:
                    return XML;
                case 3:
                    return JSON;
                case 4:
                    return HDF5;
                case 5:
                    return DBB;
                case 6:
                    return RecordSetMD;
                case 7:
                    return SQL;
                case 8:
                    return NamingConvention;
                default:
                    return Unknown;
            }
        }

        public static eSchemaType FromAlias(String arg)
        {
            switch (arg.toLowerCase()) {
                case "xsd":
                case "xml-schema":
                    return eSchemaType.XSD;
                case "xml":
                    return eSchemaType.XML;
                case "json":
                    return eSchemaType.JSON;
                case "hdf5":
                case "h5":
                    return eSchemaType.HDF5;
                case "xdatadbb":
                case "bytebuffer":
                case "bb":
                case "byte":
                case "bytes":
                    return eSchemaType.DBB;
                case "xdatamap":
                    return eSchemaType.RecordSetMD;
                case "sql":
                    return eSchemaType.SQL;
                case "namingconvention":
                case "name":
                case "names":
                    return eSchemaType.NamingConvention;
                default:
                    return eSchemaType.Unknown;
            }
        }

        @Override
        public String toString()
        {
            switch (this) {
                case XSD:
                    return "XSD";
                case XML:
                    return "XML Signature";
                case JSON:
                    return "JSON";
                case HDF5:
                    return "HDF5 Metadata";
                case DBB:
                    return "DBB Layout";
                case RecordSetMD:
                    return "Internally generated RecordSetMD";
                case SQL:
                    return "SQL";
                case NamingConvention:
                    return "NamingConvention";
                default:
                    return "Unknown";
            }
        }
    }


}

File: JniPMML-Cs/src/RecordSetMDEnums.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.WDataSci.JniPMML {

    public class RecordSetMDEnums {

        public enum eType : int { Unknown = 0, Dlm, CSV, TXT, HDF5, DBB, XML, JSON, SQL }

        public enum eMode : int { Unknown = 0, Input, Output, Internal }

        public enum eSchemaType : int { Unknown = 0, XSD, XML, JSON, HDF5, DBB, XDataMap, SQL, NamingConvention }

    }

    public static class RecordSetMDExt {

        ///////////////////////////////////////////////////////////////////////

        public static RecordSetMDEnums.eType eType_FromInt(int arg) {
            switch ( arg ) {
                case 1: return RecordSetMDEnums.eType.Dlm;
                case 2: return RecordSetMDEnums.eType.CSV;
                case 3: return RecordSetMDEnums.eType.TXT;
                case 4: return RecordSetMDEnums.eType.HDF5;
                case 5: return RecordSetMDEnums.eType.DBB;
                case 6: return RecordSetMDEnums.eType.XML;
                case 7: return RecordSetMDEnums.eType.JSON;
                case 8: return RecordSetMDEnums.eType.SQL;
                default: return RecordSetMDEnums.eType.Unknown;
            }
        }

        public static Boolean isFlatFile(this RecordSetMDEnums.eType self)
        {
            return self.bIn(RecordSetMDEnums.eType.Dlm, RecordSetMDEnums.eType.CSV, RecordSetMDEnums.eType.TXT, RecordSetMDEnums.eType.XML, RecordSetMDEnums.eType.JSON);
        }

        public static int AsInt(this RecordSetMDEnums.eType self) {
            switch ( self ) {
                case RecordSetMDEnums.eType.Dlm: return 1;
                case RecordSetMDEnums.eType.CSV: return 2;
                case RecordSetMDEnums.eType.TXT: return 3;
                case RecordSetMDEnums.eType.HDF5: return 4;
                case RecordSetMDEnums.eType.DBB: return 5;
                case RecordSetMDEnums.eType.XML: return 6;
                case RecordSetMDEnums.eType.JSON: return 7;
                case RecordSetMDEnums.eType.SQL: return 8;
                default: return 0;
            }
        }


        public static RecordSetMDEnums.eType eType_FromAlias(String arg) {
            switch ( arg.ToLower() ) {
                case "csv":
                case "comma-delimited":
                case "comma-separated":
                    return RecordSetMDEnums.eType.CSV;
                case "txt":
                case "tab-delimited":
                case "tab-separated":
                    return RecordSetMDEnums.eType.TXT;
                case "dlm":
                case "pipe":
                case "pipe-delimited":
                case "pipe-separated":
                case "delimited":
                    return RecordSetMDEnums.eType.Dlm;
                case "hdf5":
                case "h5":
                    return RecordSetMDEnums.eType.HDF5;
                case "xdatadbb":
                case "bytebuffer":
                case "bb":
                case "byte":
                case "bytes":
                    return RecordSetMDEnums.eType.DBB;
                case "json":
                    return RecordSetMDEnums.eType.JSON;
                case "xml":
                    return RecordSetMDEnums.eType.XML;
                case "sql":
                    return RecordSetMDEnums.eType.SQL;
                default:
                    return RecordSetMDEnums.eType.Unknown;
            }
        }

        public static Boolean equals(this RecordSetMDEnums.eType self, RecordSetMDEnums.eType arg) { return self.Equals(arg); }

        public static Boolean bIn(this RecordSetMDEnums.eType self, params RecordSetMDEnums.eType[] args) {
            foreach ( RecordSetMDEnums.eType arg in args ) if ( self.Equals(arg) ) return true;
            return false;
        }

        public static Boolean bIn(this RecordSetMDEnums.eType self, params String[] args) {
            foreach ( String s in args ) if ( self.Equals(eType_FromAlias(s)) ) return true;
            return false;
        }

        public static String toString(this RecordSetMDEnums.eType self)
        {
            switch (self) {
                case RecordSetMDEnums.eType.Dlm:
                    return "Dlm";
                case RecordSetMDEnums.eType.CSV:
                    return "CSV";
                case RecordSetMDEnums.eType.TXT:
                    return "TXT";
                case RecordSetMDEnums.eType.HDF5:
                    return "HDF5";
                case RecordSetMDEnums.eType.DBB:
                    return "DBB";
                case RecordSetMDEnums.eType.XML:
                    return "XML";
                case RecordSetMDEnums.eType.JSON:
                    return "JSON";
                case RecordSetMDEnums.eType.SQL:
                    return "SQL";
                default:
                    return "Unknown";
            }
        }



        ///////////////////////////////////////////////////////////////////////

        public static RecordSetMDEnums.eMode eMode_FromInt(int arg) {
            switch ( arg ) {
                case 1: return RecordSetMDEnums.eMode.Input;
                case 2: return RecordSetMDEnums.eMode.Output;
                case 3: return RecordSetMDEnums.eMode.Internal;
                default: return RecordSetMDEnums.eMode.Unknown;
            }
        }


        public static int AsInt(this RecordSetMDEnums.eMode self) {
            switch ( self ) {
                case RecordSetMDEnums.eMode.Input: return 1;
                case RecordSetMDEnums.eMode.Output: return 2;
                case RecordSetMDEnums.eMode.Internal: return 3;
                default: return 0;
            }
        }


        public static RecordSetMDEnums.eMode eMode_FromAlias(String arg) {
            switch ( arg.ToLower() ) {
                case "input":
                case "in":
                    return RecordSetMDEnums.eMode.Input;
                case "output":
                case "out":
                    return RecordSetMDEnums.eMode.Output;
                case "internal":
                case "int":
                    return RecordSetMDEnums.eMode.Internal;
                default:
                    return RecordSetMDEnums.eMode.Unknown;
            }
        }

        public static Boolean equals(this RecordSetMDEnums.eMode self, RecordSetMDEnums.eMode arg) { return self.Equals(arg); }

        public static Boolean bIn(this RecordSetMDEnums.eMode self, params RecordSetMDEnums.eMode[] args) {
            foreach ( RecordSetMDEnums.eMode arg in args ) if ( self.Equals(arg) ) return true;
            return false;
        }

        public static Boolean bIn(this RecordSetMDEnums.eMode self, params String[] args) {
            Boolean found=false;
            foreach ( String s in args ) if ( self.Equals(eMode_FromAlias(s)) ) return true;
            return false;
        }



        ///////////////////////////////////////////////////////////////////////


        public static RecordSetMDEnums.eSchemaType eSchemaType_FromInt(int arg) {
            switch ( arg ) {
                case 1: return RecordSetMDEnums.eSchemaType.XSD;
                case 2: return RecordSetMDEnums.eSchemaType.XML;
                case 3: return RecordSetMDEnums.eSchemaType.JSON;
                case 4: return RecordSetMDEnums.eSchemaType.HDF5;
                case 5: return RecordSetMDEnums.eSchemaType.DBB;
                case 6: return RecordSetMDEnums.eSchemaType.XDataMap;
                case 7: return RecordSetMDEnums.eSchemaType.SQL;
                case 8: return RecordSetMDEnums.eSchemaType.NamingConvention;
                default: return RecordSetMDEnums.eSchemaType.Unknown;
            }
        }


        public static int AsInt(this RecordSetMDEnums.eSchemaType self) {
            switch ( self ) {
                case RecordSetMDEnums.eSchemaType.XSD: return 1;
                case RecordSetMDEnums.eSchemaType.XML: return 2;
                case RecordSetMDEnums.eSchemaType.JSON: return 3;
                case RecordSetMDEnums.eSchemaType.HDF5: return 4;
                case RecordSetMDEnums.eSchemaType.DBB: return 5;
                case RecordSetMDEnums.eSchemaType.XDataMap: return 6;
                case RecordSetMDEnums.eSchemaType.SQL: return 7;
                case RecordSetMDEnums.eSchemaType.NamingConvention: return 8;
                default: return 0;
            }
        }

        public static Boolean equals(this RecordSetMDEnums.eSchemaType self, RecordSetMDEnums.eSchemaType arg) { return self.Equals(arg); }

        public static RecordSetMDEnums.eSchemaType eSchemaType_FromAlias(String arg) {
            switch ( arg.ToLower() ) {
                case "xsd":
                case "xmlschema":
                    return RecordSetMDEnums.eSchemaType.XSD;
                case "xml":
                    return RecordSetMDEnums.eSchemaType.XML;
                case "json":
                    return RecordSetMDEnums.eSchemaType.JSON;
                case "hdf5":
                    return RecordSetMDEnums.eSchemaType.HDF5;
                case "xdatadbb":
                case "bytebuffer":
                case "bb":
                case "byte":
                case "bytes":
                    return RecordSetMDEnums.eSchemaType.DBB;
                case "xdatamap":
                    return RecordSetMDEnums.eSchemaType.XDataMap;
                case "sql":
                    return RecordSetMDEnums.eSchemaType.SQL;
                case "namingconvention":
                case "name":
                case "names":
                    return RecordSetMDEnums.eSchemaType.NamingConvention;
                default:
                    return RecordSetMDEnums.eSchemaType.Unknown;
            }
        }

        public static String toString(this RecordSetMDEnums.eSchemaType self)
        {
            switch (self) {
                case RecordSetMDEnums.eSchemaType.XSD:
                    return "XSD";
                case RecordSetMDEnums.eSchemaType.XML:
                    return "XML Signature";
                case RecordSetMDEnums.eSchemaType.JSON:
                    return "JSON";
                case RecordSetMDEnums.eSchemaType.HDF5:
                    return "HDF5 Metadata";
                case RecordSetMDEnums.eSchemaType.DBB:
                    return "DBB Layout";
                case RecordSetMDEnums.eSchemaType.XDataMap:
                    return "Internally generated RecordSetMD";
                case RecordSetMDEnums.eSchemaType.SQL:
                    return "SQL";
                case RecordSetMDEnums.eSchemaType.NamingConvention:
                    return "NamingConvention";
                default:
                    return "Unknown";
            }
        }
    
        public static Boolean bIn(this RecordSetMDEnums.eSchemaType self, params RecordSetMDEnums.eSchemaType[] args) {
            foreach ( RecordSetMDEnums.eSchemaType arg in args ) if ( self.Equals(arg) ) return true;
            return false;
        }

        public static Boolean bIn(this RecordSetMDEnums.eSchemaType self, params String[] args) {
            foreach ( String s in args ) if ( self.Equals(eSchemaType_FromAlias(s)) ) return true;
            return false;
        }

    }
}


